homework-maven-plugin/src/main/java/hh/TestFiles.java

```diff
@@ -0,0 +1,70 @@
+package hh;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.io.File;
+import java.util.ArrayList;
+
+/**
+ * Goal which scan dir and check files
+ */
+@Mojo(name = "scanAndTest", defaultPhase = LifecyclePhase.PROCESS_SOURCES)
+public class TestFiles extends AbstractMojo {
+
+    private void recursiveDirScan(File start, ArrayList<File> filelist) {
+        try {
+            if (start.isDirectory()) {
```

Comment:
методы нужно определять после параметров

и лучше переделать примерно таким образом
```suggestion
        private List<File> recursiveDirScan(File start) {
        if (!start.isDirectory()) {
            return Collections.emptyList();
        }
        List<File> result = new ArrayList<>();
        for (File item : Objects.requireNonNull(start.listFiles())) {
            if (item.isDirectory()) {
                result.addAll(recursiveDirScan(item));
            } else {
                result.add(item);
            }
        }
        return result;
    }
```

не очень хорошо, что изменяется входной параметр

Reply:
А вообще можно это сделать с помощью стандартных функций. 
Посмотри соседние PR